marketplaceItems/audioFocus/appResources/appData/audioFocus_app.js

```diff
@@ -0,0 +1,459 @@
+
+        var avatarIntersection = AvatarList.findRayIntersection({origin:start, direction:direction});
+        
+        var uuid = avatarIntersection.avatarID;
+
+        return uuid;
+    }
+
+
+    var MAPPING_NAME = "SOLO_POINTER";
+    var mapping = Controller.newMapping(MAPPING_NAME);
+
+    mapping.from(Controller.Hardware.Keyboard.LeftMouseButton).to(function (value) {
+        if (value === 1 || HMD.active) {
+            return;
+        }
+
+        var pickRay = Camera.computePickRay(lastMouseX, lastMouseY);
+        var avatarIntersection = AvatarList.findRayIntersection(pickRay);
+
+        var uuid = avatarIntersection.avatarID;
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(avatarIntersection.avatarID);
+        }
+    });
+
+
+    mapping.from(Controller.Standard.LTClick).to(function (value) {
+        if (value === 0) {
+            return;
+        }
+        
+        var uuid = getUUIDFromLaser(Controller.Standard.LeftHand);
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(uuid);
+        }
+    });
+
+
+    mapping.from(Controller.Standard.RTClick).to(function (value) {
+        if (value === 0) {
+            return;
+        }
+        
+        var uuid = getUUIDFromLaser(Controller.Standard.RightHand);
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(uuid);
+        }
+    });
+
+
+    // #endregion
+    // *************************************
+    // STOP MAPPING FUNCTIONS
+    // *************************************
+
+
+    // *************************************
+    // START SOLO FUNCTIONS
+    // *************************************
+    // #region Solo
+
+
+    // Adds avatar to the solo list
+    function addSolo(targetUUID) {
+        Audio.addToSoloList([targetUUID]);
+        updateUI();
+    }
+
+
+    // Remove Avatar from the solo list
+    function removeSolo(targetUUID) {
+        Audio.removeFromSoloList([targetUUID]);
+        updateUI();
+    }
+
+
+    // Remove all avatars from the solo list
+    function resetSolo() {
+        Audio.resetSoloList();
+        removeAllOverlays();
+        soloAvatars = {};
+        updateUI();
+    }
+
+    
+    // Add an avatar to the list and give them an overlay
+    function addAvatarToList(avatarUUID, displayUsername) {
+        soloAvatars[avatarUUID] = {
+            id: avatarUUID,
+            name: displayUsername
+        };
+
+        addSolo(avatarUUID);
+        addOverlayToUser(avatarUUID);
+    }
+
+
+    // Remove an avatar from the list and remove their overlay
+    function removeAvatarFromList(avatarUUID) {
+        removeOverlay(avatarUUID);
+        delete soloAvatars[avatarUUID];
+        removeSolo(avatarUUID);
+    }
+
+
+    // Handles avatar being solo'd
+    function soloAvatar(avatarUUID) {
+        log("in soloAvatar", avatarUUID);
+
+        log("soloAvatars", soloAvatars);
```

Comment:
(i know this is only here for debug during prototype, but i like having comments about things we've talked about before on new commits when the old comments are dismissed)